stdlib/public/Reflection/TypeRef.cpp

```diff
@@ -715,6 +763,74 @@ class DemanglingForTypeRef
     return node;
   }
 
+  Demangle::NodePointer
+  visitSILBoxTypeWithLayoutTypeRef(const SILBoxTypeWithLayoutTypeRef *SB) {
+    auto node = Dem.createNode(Node::Kind::SILBoxTypeWithLayout);
+    auto layout = Dem.createNode(Node::Kind::SILBoxLayout);
+    for (auto &f : SB->getFields()) {
+      auto field =
+          Dem.createNode(f.isMutable() ? Node::Kind::SILBoxMutableField
+                                       : Node::Kind::SILBoxImmutableField);
+      field->addChild(visit(f.getType()), Dem);
+      layout->addChild(field, Dem);
+    }
+    node->addChild(layout, Dem);
+
+    auto signature = Dem.createNode(Node::Kind::DependentGenericSignature);
+    auto addCount = [&](unsigned count) {
+      signature->addChild(
+          Dem.createNode(Node::Kind::DependentGenericParamCount, count), Dem);
+    };
+    unsigned depth = 0;
+    unsigned index = 0;
+    for (auto &s : SB->getSubstitutions())
+      if (auto *param = dyn_cast<GenericTypeParameterTypeRef>(s.first)) {
+        while (param->getDepth() > depth) {
+          addCount(index);
+          ++depth, index = 0;
+        }
+        assert(index == param->getIndex() && "generic params out of order");
+        ++index;
+      }
+    for (auto &req : SB->getRequirements()) {
+      switch (req.getKind()) {
+      case RequirementKind::Conformance:
+      case RequirementKind::Superclass:
+      case RequirementKind::SameType: {
+        Node::Kind kind;
+        switch (req.getKind()) {
+        case RequirementKind::Conformance:
+          kind = Node::Kind::DependentGenericConformanceRequirement;
+          break;
+        case RequirementKind::Superclass:
+          // Is this node kind missing, or is this impossible?
```

Comment:
It's certainly possible to have a class-constrained generic parameter...

Reply:
I changed the comment.